src/sentry/api/event_search.py

```diff
@@ -83,7 +84,9 @@ def translate(pat):
 
 
 event_search_grammar = Grammar(r"""
-search          = search_term*
+search          = (nested_boolean_term / boolean_term / search_term)*
+nested_boolean_term = boolean_term boolean_operator search_term
+boolean_term          = search_term boolean_operator search_term
```

Comment:
Does this only enable 1 level of nesting? Should we support more than one level?

Reply:
oops you're right there's a typo there. Good catch. I'll add a test to ensure that works.